examples/semantic_segmentation/main.js

```diff
@@ -1,556 +1,207 @@
+let utils = new Utils();
+utils.updateProgress = updateProgress;
+
+const updateResult = (result) => {
+  try {
+    console.log(`Inference time: ${result.time} ms`);
+    let inferenceTimeElement = document.getElementById('inferenceTime');
+    inferenceTimeElement.innerHTML = `inference time: <span class='ir'>${result.time} ms</span>`;
+  } catch (e) {
+    console.log(e);
+  }
+  try {
+    console.log(`Classes: `);
+    result.classes.forEach((c, i) => {
+      console.log(`\tlabel: ${c.label}, probability: ${c.prob}%`);
+      let labelElement = document.getElementById(`label${i}`);
+      let probElement = document.getElementById(`prob${i}`);
+      labelElement.innerHTML = `${c.label}`;
+      probElement.innerHTML = `${c.prob}%`;
+    });
   }
-
-
-  // register backends
-  if (nnNative) {
-    webml.setAttribute('class', 'dropdown-item');
-    webml.onclick = function (e) {
-      removeAlertElement();
-      checkPreferParam();
-      changeBackend('WebML');
-    };
+  catch (e) {
+    console.log(e);
   }
+}
 
-  if (nnPolyfill.supportWebGL) {
-    webgl.setAttribute('class', 'dropdown-item');
-    webgl.onclick = function (e) {
-      removeAlertElement();
-      changeBackend('WebGL');
-    };
-  }
+const logConfig = () => {
+  console.log(`Model: ${currentModel}, Backend: ${currentBackend}, Prefer: ${currentPrefer}`);
+}
 
-  if (nnPolyfill.supportWasm) {
-    wasm.setAttribute('class', 'dropdown-item');
-    wasm.onclick = function (e) {
-      removeAlertElement();
-      changeBackend('WASM');
-    };
-  }
+const errorHandler = (e) => {
+  showAlert(e);
+  showError(null, null);
+}
 
-  if (currentBackend === '') {
-    if (nnNative) {
-      currentBackend = 'WebML';
-    } else {
-      currentBackend = 'WebGL';
+const startPredict = async () => {
+  if (streaming) {
+    try {
+      stats.begin();
+      await predictAndDraw(videoElement, true);
+      stats.end();
+      setTimeout(startPredict, 0);
+    } catch (e) {
+      errorHandler(e);
     }
   }
+}
 
-  function setCamResolution(resolution) {
-    return navigator.mediaDevices.getUserMedia({
-      audio: false,
-      video: { facingMode: 'user' }
-    }).then((stream) => {
-      videoElement.srcObject = stream;
-      return new Promise((resolve) => {
-        // video cannot be uploaded to texture until being loaded
-        videoElement.onloadeddata = resolve;
-      });
-    }).catch((error) => {
-      console.log('getUserMedia error: ' + error.name, error);
+const predictCamera = async () => {
+  try {
+    streaming = true;
+    // let res = utils.getFittedResolution(4 / 3);
+    // setCamResolution(res);
+    let stream = await navigator.mediaDevices.getUserMedia({ audio: false, video: { facingMode: 'user' } });
+    videoElement.srcObject = stream;
+    track = stream.getTracks()[0];
+    showProgress('Inferencing ...');
+    return new Promise((resolve) => {
+      videoElement.onloadeddata = resolve;
+      startPredict();
```

Comment:
Camera predict should start after `onloadeddata`. Otherwise, a lot of warnings about video texture will show in the beginning. I would suggest the following instead.
```js
const predictCamera = async () => {
  try {
    streaming = true;
    let stream = await navigator.mediaDevices.getUserMedia({ audio: false, video: { facingMode: 'user' } });
    videoElement.srcObject = stream;
    track = stream.getTracks()[0];
    showProgress('Inferencing ...');
    videoElement.onloadeddata = startPredict;
  } catch (e) {
    errorHandler(e);
  }
}
```

Reply:
@pinzhenx Thanks for the great suggestion, please review the update code in https://github.com/intel/webml-polyfill/pull/560/commits/aaa7dee1d7745e96785509a172825ccd4fb04cd2

Reply:
Thanks. No more warnings.